node/src/components/consensus/consensus_protocol.rs

```diff
@@ -109,6 +111,57 @@ pub struct EraReport<VID> {
     pub(crate) inactive_validators: Vec<VID>,
 }
 
+impl<VID> Default for EraReport<VID>
+where
+    VID: Ord,
+{
+    fn default() -> Self {
+        EraReport {
+            equivocators: vec![],
+            rewards: BTreeMap::new(),
+            inactive_validators: vec![],
+        }
+    }
+}
+
+impl<VID> EraReport<VID> {
+    pub fn hash(&self) -> Digest
+    where
+        VID: ToBytes,
+    {
+        // Helper function to hash slice of validators
+        fn hash_slice_of_validators<VID>(slice_of_validators: &[VID]) -> Digest
+        where
+            VID: ToBytes,
+        {
+            let hashes = slice_of_validators
+                .iter()
+                .map(|validator| {
+                    hash::hash(validator.to_bytes().expect("Could not serialize validator"))
+                })
+                .collect();
+            hash::hash_vec_merkle_tree(hashes)
+        }
+
+        // Pattern match here leverages compiler to ensure every field is accounted for
+        let EraReport {
+            equivocators,
+            inactive_validators,
+            rewards,
+        } = self;
+
+        let hashed_equivocators = hash_slice_of_validators(equivocators);
+        let hashed_inactive_validators = hash_slice_of_validators(inactive_validators);
+        let hashed_rewards = hash::hash_btree_map(rewards).expect("Could not hash rewards");
+
+        hash::hash_slice_rfold(&[
+            hashed_equivocators,
+            hashed_rewards,
+            hashed_inactive_validators,
```

Comment:
It's a bit weird to put the rewards in the middle. Maybe first equivocators, then inactive validators, then rewards? (But then we should probably change that everywhere in the code, too, so it doesn't have to be in this PR.)